amida-tech/amida-auth-microservice
# Logging updates

If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
* Adds swaggerstats for default instrumentation
* Updates sequelize to use winston for logging so that format is consistent
* Updated winston to 3.0.0 (for improved json logging support)
* Small adjustments to dockerfile to use a multi-stage build
one deficiency to this PR is that swagger-stats is looking for a favicon.ico, I would like to see one added but do not have one.

#### Related JIRA tickets:
Orange-624
#### How should this be manually tested?
* Run the application, logging outside of the sequelize warning should be json formatted.
* go to <address>:<port>/swagger-stats/ui and confirm you see swagger stats
* go to <address>:<port>/swagger-stats/metrics and confirm you see prometheus configured metrics
* `docker build -t my-auth-test .` and confirm it builds
#### Any background context you want to provide?

This commit is also a stepping stone so I can add metrics on system health to the prometheus endpoint (allowing for more comprehensive consolidated status)
#### Screenshots (if appropriate):
